06-building-a-program-start-to-finish/solving-the-problem/Output/model/inverted-index.js

```diff
@@ -0,0 +1,168 @@
+    for (let i = 0; i < titles.length; i += 1) {
+      documents.push(i);
+    }
+    this.allIndexed[fileName] = indexed;
+    this.allTitles[fileName] = titles;
+    this.documentsIndex[fileName] = documents;
+    return indexed;
+  }
+
+  /**
+   * Get Index
+   * getIndex method gets the indexed file with words from documents that were
+   * found. If the file has not been indexed, it calls the create index method
+   * to create the index
+   * @param {Object} fileContent - the file content we want to get indexed
+   * @param {String} fileName - the name of the file to be indexed
+   * @return {Object} all words in the file and their corresponding indexes
+   */
+  getIndex(fileContent, fileName) {
+    if (fileName in this.allIndexed) {
+      return this.allIndexed[fileName];
+    }
+    return false;
+  }
+
+   /**
+   * tokenize method removes special characters and returns an array of words
+   * @param {string} text - the validated text to be tokenized
+   * @return {array} array of words in the document
+  */
+  tokenize(text) {
+    this.text = text;
+    const cleanWords = this.text.toLowerCase().match(/[a-z0-9]+/g);
+    return cleanWords;
+  }
+
+  /**
+   * validateFile method ensures all the documents in a file are valid i.e.
+   * they should be arrays of objects that have only title and text keys
+   * and Strings as value
+   * @param {Object} fileContent - the uploaded contents of the file
+   * @return {Boolean} true or false if the method was successful or not
+  */
+  validateFile(fileContent) {
+    this.fileContent = fileContent;
+    if (Array.isArray(this.fileContent)) {
+      for (let item = 0; item < this.fileContent.length; item += 1) {
+        if ((typeof this.fileContent[item]) !== 'object') {
+          return false;
+        }
+        if (Object.keys(this.fileContent[item]).length !== 2 ||
+            Object.keys(this.fileContent[item])[0] !== 'title' ||
+            Object.keys(this.fileContent[item])[1] !== 'text') {
+          return false;
+        }
+        if ((typeof this.fileContent[item].title) !== 'string' ||
+            (typeof this.fileContent[item].text) !== 'string') {
+          return false;
+        }
+      }
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * readFile method reads the data from the file being uploaded
+   * @param {JSON} rawFile - the raw uploaded file
+   * @return {String} string of text in the uploaded file
+  */
+  readFile(rawFile) {
+    this.rawFile = rawFile;
+    return new Promise((resolve, reject) => {
+      const reader = new FileReader();
+      reader.onload = data => resolve(data.target.result);
+      reader.onerror = error => reject(`Error reading + 
+                                ${this.rawFile.name}: ${error.target.result}`);
+      reader.readAsText(this.rawFile);
+    });
+  }
+
+  /**
+   * searchIndex method searches the indexed files for occurences of words
+   * @param {String} word - word(s) that one is searching for
```

Comment:
Is your searchIndex() accepting one _word_ per search or _words_? I believe your parameter names can be a lot more descriptive.